server_list/ping.py

```diff
@@ -0,0 +1,175 @@
+import asyncio
+import time
+import random
+import socket
+
+from .app import app
+from .util import get_addr_info
+
+
+class MinetestProtocol:
+	def connection_made(self, transport):
+		self.transport = transport
+
+	def send_original(self):
+		# Send packet of type ORIGINAL, with no data.
+		# This should prompt the server to assign us a peer id.
+		# [0] u32       protocol_id (PROTOCOL_ID)
+		# [4] session_t sender_peer_id (PEER_ID_INEXISTENT)
+		# [6] u8        channel
+		# [7] u8        type (PACKET_TYPE_ORIGINAL)
+		self.transport.sendto(b"\x4f\x45\x74\x03\x00\x00\x00\x01")
```

Comment:
sendto missing address

Reply:
It's optional.  There's no `send()`, just a `sendto()` with an optional address (the transport has an associated address).  Bit of an odd API choice, but that's how it is: https://docs.python.org/3/library/asyncio-protocol.html#asyncio.DatagramTransport.sendto

Reply:
Why does the other `sendto` call have an address then?